doc/doxygen/scripts/set_canonical_doxygen.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2019 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+# This script sets the canonical webpage for all the webpages of the
+# doxygen documentation. For more information see
+# https://en.wikipedia.org/wiki/Canonical_link_element
+#
+# The script should be invoked from the root folder of the doxygen
+# documentation, for example:
+# cd build/doc/doxygen/deal.II/
+# dealii_repository/doc/doxygen/scripts/set_canonical_doxygen.py
+
+import glob
+
+filenames = glob.iglob('**/*html', recursive=True)
+
+for filename in filenames:
+    file = open(filename, 'r')
+    new_text_data = str()
+    if '<link rel="canonical"' not in file.read():
+        file.seek(0)
+        for line in file.readlines():
+            # Do not add the canonical link twice
+            canonical_link_added = False
+            new_text_data += line
+            if (not canonical_link_added) and ('<head>' in line):
+                new_text_data += ('<link rel="canonical" href="https://www.dealii.org/current/doxygen/deal.II/'
```

Comment:
@tamiko I saw in #7636 that you are going to apply this script to the old documentation. Note that this script should be run in `doc/doxygen/deal.II/` and will only add the canonical link to the doxygen documentation, it will not add the link to the rest of the documentation.

If you want to add the link to all the old documentation the script should be run in `doc` and the line should be replaced by
``` python
                new_text_data += ('<link rel="canonical" href="https://www.dealii.org/current/'
```